### PR TITLE
scripts: west_commands: runners: Add support for probe-rs

### PR DESCRIPTION
**Pull request by xingrz**
[probe-rs](https://probe.rs/) is a new programming and debugging tool written in Rust, supports many probes and targets.

This PR introduces initial support for probe-rs to Zephyr.

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for contributing this probe.
Please update doc/develop/flash_debug/host-tools.rst, which will also be useful for review and testing

**Comment by jaredwolff** [comment]
It would be great to see this go through. Was there anything else you wanted to see in `host-tools.rst` @erwango?

**Review by kartben — CHANGES_REQUESTED** [review]
a small typo

File: doc/develop/flash_debug/host-tools.rst [hunk 1]
```diff
@@ -460,6 +460,17 @@ afterwards detach the debug session:
 
    west debug --tool-opt='--batch'
 
+probe-rs Debug Host Tools
+*************************
+
+probe-rs is an open-source emmbedded toolkit written in Rust. It provides
```
[the remark this answers is not in the document]
Comment (xingrz) [reply doc/develop/flash_debug/host-tools.rst]: Thank you

**Review by kartben — APPROVED** [review]

**Review by erwango — APPROVED** [review]
Txs for the update

**Review by carlescufi — APPROVED** [review]

**Pull request merged**